homeassistant/components/tplink/__init__.py

```diff
@@ -83,8 +84,13 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     except SmartDeviceException as ex:
         raise ConfigEntryNotReady from ex
 
-    hass.data[DOMAIN][entry.entry_id] = TPLinkDataUpdateCoordinator(hass, device)
+    entry.options.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL_DEFAULT)
```

Comment:
Manually controlling the polling interval can be done for any integration already. This built-in.

You can disable polling on any config entry and at that point use the `homeassistant.update_entity` service to define your own schedule, interval, etc for updating and polling.

Since its introduction, we thus no longer allow these settings (as it is a duplicate functionality).

../Frenck

Reply:
Hi @frenck!

I feel kind of stupid. Although I managed to add this what appears to be duplicate functionality I have no idea what you are referring to. Any guidance on how to "disable polling on config entry" and how to use `homeassistant.update_entity` to define polling interval?

Reply:
Click on the three-dotted menu on an integration card for the entry you want to poll manually, and select "System options".

Next, in the dialog shown, disable the "Polling for update" for the integration.

Create an automation with your wished-for pattern and conditions calling the `homeassistant.update_entity` service, targeting one of the entities the integration provided.

You can also leave the existing polling enabled, in that case, the service calls made with `homeassistant.update_entity` are additional to the existing default interval.